### PR TITLE
planner: use code-gen to generate CloneForPlanCache method for some operators

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #54057

Problem Summary: planner: use code-gen to generate CloneForPlanCache method for some operators

### What changed and how does it work?

planner: use code-gen to generate CloneForPlanCache method for some operators

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: pkg/planner/core/physical_plans.go [hunk 1]
```diff
@@ -430,12 +430,12 @@ func (p *PushedDownLimit) MemoryUsage() (sum int64) {
 type PhysicalIndexLookUpReader struct {
 	physicalSchemaProducer
 
+	indexPlan  base.PhysicalPlan
+	tablePlan  base.PhysicalPlan
```
Comment (AilinKid) [inline pkg/planner/core/physical_plans.go]: better to export for later pkg move

File: pkg/planner/core/plan_clone_generated.go [hunk 2]
```diff
@@ -180,3 +180,104 @@ func (op *PhysicalMergeJoin) CloneForPlanCache(newCtx base.PlanContext) (base.Pl
...
+	}
+	cloned.physicalSchemaProducer = *basePlan
+	tablePlan, ok := op.tablePlan.CloneForPlanCache(newCtx)
+	if !ok {
+		return nil, false
+	}
+	cloned.tablePlan = tablePlan.(base.PhysicalPlan)
+	cloned.TablePlans = flattenPushDownPlan(cloned.tablePlan)
+	cloned.PlanPartInfo = op.PlanPartInfo.Clone()
+	if op.TableScanAndPartitionInfos != nil {
```
Comment (AilinKid) [inline pkg/planner/core/plan_clone_generated.go:200]: if TableScanAndPartitionInfos != nil, refuse clone?

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

File: pkg/planner/core/plan_clone_generated.go (hunk 2 above)
Comment (qw4990) [reply pkg/planner/core/plan_clone_generated.go:200]: YES, this field is only for MPP, but MPP plan is not supported by Plan Cache.

**Review by AilinKid — APPROVED** [review]

**Pull request merged**